pkg/dockerregistry/server/imagestream.go

```diff
@@ -170,3 +171,39 @@ func (is *imageStream) rememberLayersOfImage(ctx context.Context, image *imageap
 		_ = is.cache.AddDigest(ref.Digest, cacheName)
 	}
 }
+
+func (is *imageStream) getSecrets() ([]kapiv1.Secret, error) {
+	secrets, err := is.registryOSClient.ImageStreamSecrets(is.namespace).Secrets(is.name, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error getting secrets for repository %s: %v", is.Reference(), err)
+	}
+	return secrets.Items, nil
+}
+
+// tagIsInsecure returns true if the given image stream or its tag allow for
+// insecure transport.
+func (is *imageStream) tagIsInsecure(tag string, dgst digest.Digest) (bool, error) {
+	stream, err := is.imageStreamGetter.get()
+	if err != nil {
+		return false, err
+	}
+
+	if len(tag) == 0 {
+		// if the client pulled by digest, find the corresponding tag in the image stream
+		tag, _ = imageapiv1.LatestImageTagEvent(stream, dgst.String())
+	}
+
+	if insecure, _ := stream.Annotations[imageapi.InsecureRepositoryAnnotation]; insecure == "true" {
```

Comment:
Looks like this condition uses only `stream`. Please move it before `if len(tag) == 0`.